File: swagger-codegen/csharp-dotnet2/src/main/CsharpDotNet2/samsara/Model/DocumentBase.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace samsara.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class DocumentBase {
    /// <summary>
    /// ID of the Samsara dispatch job for which the document is submitted
    /// </summary>
    /// <value>ID of the Samsara dispatch job for which the document is submitted</value>
    [DataMember(Name="dispatchJobId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "dispatchJobId")]
    public long? DispatchJobId { get; set; }

    /// <summary>
    /// Notes submitted with this document.
    /// </summary>
    /// <value>Notes submitted with this document.</value>
    [DataMember(Name="notes", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "notes")]
    public string Notes { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class DocumentBase {\n");
      sb.Append("  DispatchJobId: ").Append(DispatchJobId).Append("\n");
      sb.Append("  Notes: ").Append(Notes).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-generator/dart/lib/model/document_base.dart
part of openapi.api;

class DocumentBase {
  /* ID of the Samsara dispatch job for which the document is submitted */
  int dispatchJobId = null;
  /* Notes submitted with this document. */
  String notes = null;
  DocumentBase();

  @override
  String toString() {
    return 'DocumentBase[dispatchJobId=$dispatchJobId, notes=$notes, ]';
  }

  DocumentBase.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['dispatchJobId'] == null) {
      dispatchJobId = null;
    } else {
          dispatchJobId = json['dispatchJobId'];
    }
    if (json['notes'] == null) {
      notes = null;
    } else {
          notes = json['notes'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'dispatchJobId': dispatchJobId,
      'notes': notes
    };
  }

  static List<DocumentBase> listFromJson(List<dynamic> json) {
    return json == null ? new List<DocumentBase>() : json.map((value) => new DocumentBase.fromJson(value)).toList();
  }

  static Map<String, DocumentBase> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, DocumentBase>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new DocumentBase.fromJson(value));
    }
    return map;
  }
}

